src/prefect/workers/base.py

```diff
@@ -950,3 +952,26 @@ def _emit_flow_run_submitted_event(self, configuration: BaseJobConfiguration):
             resource=self._event_resource(),
             related=related,
         )
+
+    def _emit_flow_run_monitored_event(
+        self, result: BaseWorkerResult, configuration: BaseJobConfiguration
+    ):
+        related = configuration._related_resources()
+
+        if self._work_pool:
+            related.append(
+                object_as_related_resource(
+                    kind="work-pool", role="work-pool", object=self._work_pool
+                )
+            )
+
+        for resource in related:
+            if resource.role == "flow-run":
+                resource["prefect.infrastructure.identifier"] = str(result.identifier)
+                resource["prefect.infrastructure.status-code"] = str(result.status_code)
+
+        emit_event(
+            event=f"prefect.worker.monitored-flow-run",
```

Comment:
`monitored-flow-run` sounds a bit like a heartbeat event, doesn't it?  I know naming these is painful.  What about `concluded` or `finalized` or `returned` or something else that has a flavor of a thing ending but not commenting on what it's state is?

Reply:
I discussed this with @chrisguidry and we decided to go with `executed-flow-run` and he mentioned that we should be setting the `follows`, which is done now as well.